src/test/TestGhoBucketCapacitySteward.t.sol

```diff
@@ -0,0 +1,212 @@
+
+contract TestGhoBucketCapacitySteward is TestGhoBase {
+  address internal NEW_OWNER = makeAddr('NEW_OWNER');
+
+  function setUp() public {
+    // Deploy Gho Bucket Capacity Steward
+    GHO_BUCKET_CAPACITY_STEWARD = new GhoBucketCapacitySteward(
+      SHORT_EXECUTOR,
+      address(GHO_TOKEN),
+      RISK_COUNCIL
+    );
+    address[] memory controlledFacilitators = new address[](2);
+    controlledFacilitators[0] = address(GHO_ATOKEN);
+    controlledFacilitators[1] = address(GHO_GSM);
+    vm.prank(SHORT_EXECUTOR);
+    GHO_BUCKET_CAPACITY_STEWARD.setControlledFacilitator(controlledFacilitators, true);
+
+    /// @dev Since block.timestamp starts at 0 this is a necessary condition (block.timestamp > `MINIMUM_DELAY`) for the timelocked contract methods to work.
+    vm.warp(GHO_BUCKET_CAPACITY_STEWARD.MINIMUM_DELAY() + 1);
+
+    // Grant roles
+    GHO_TOKEN.grantRole(GHO_TOKEN_BUCKET_MANAGER_ROLE, address(GHO_BUCKET_CAPACITY_STEWARD));
+  }
+
+  function testConstructor() public {
+    assertEq(GHO_BUCKET_CAPACITY_STEWARD.owner(), SHORT_EXECUTOR);
+    assertEq(GHO_BUCKET_CAPACITY_STEWARD.GHO_TOKEN(), address(GHO_TOKEN));
+    assertEq(GHO_BUCKET_CAPACITY_STEWARD.RISK_COUNCIL(), RISK_COUNCIL);
+
+    address[] memory controlledFacilitators = GHO_BUCKET_CAPACITY_STEWARD
+      .getControlledFacilitators();
+    assertEq(controlledFacilitators.length, 2);
+
+    uint40 facilitatorTimelock = GHO_BUCKET_CAPACITY_STEWARD.getFacilitatorBucketCapacityTimelock(
+      controlledFacilitators[0]
+    );
+    assertEq(facilitatorTimelock, 0);
+  }
+
+  function testRevertConstructorInvalidExecutor() public {
+    vm.expectRevert('INVALID_OWNER');
+    new GhoBucketCapacitySteward(address(0), address(0x002), address(0x003));
+  }
+
+  function testRevertConstructorInvalidGhoToken() public {
+    vm.expectRevert('INVALID_GHO_TOKEN');
+    new GhoBucketCapacitySteward(address(0x001), address(0), address(0x003));
+  }
+
+  function testRevertConstructorInvalidRiskCouncil() public {
+    vm.expectRevert('INVALID_RISK_COUNCIL');
+    new GhoBucketCapacitySteward(address(0x001), address(0x002), address(0));
+  }
+
+  function testChangeOwnership() public {
+    assertEq(GHO_BUCKET_CAPACITY_STEWARD.owner(), SHORT_EXECUTOR);
+    vm.prank(SHORT_EXECUTOR);
+    GHO_BUCKET_CAPACITY_STEWARD.transferOwnership(NEW_OWNER);
+    assertEq(GHO_BUCKET_CAPACITY_STEWARD.owner(), NEW_OWNER);
+  }
+
+  function testUpdateFacilitatorBucketCapacity() public {
+    (uint256 currentBucketCapacity, ) = GHO_TOKEN.getFacilitatorBucket(address(GHO_ATOKEN));
+    vm.prank(RISK_COUNCIL);
+    uint128 newBucketCapacity = uint128(currentBucketCapacity) + 1;
+    GHO_BUCKET_CAPACITY_STEWARD.updateFacilitatorBucketCapacity(
+      address(GHO_ATOKEN),
+      newBucketCapacity
+    );
+    (uint256 capacity, ) = GHO_TOKEN.getFacilitatorBucket(address(GHO_ATOKEN));
+    assertEq(newBucketCapacity, capacity);
+  }
+
+  function testUpdateFacilitatorBucketCapacityMaxValue() public {
```

Comment:
missing test cases for capacity decreases

Reply:
Done